src/engine/ExportQueryExecutionTrees.cpp

```diff
@@ -753,3 +863,66 @@ nlohmann::json ExportQueryExecutionTrees::computeResultAsJSON(
     throw;
   }
 }
+
+// _____________________________________________________________________________
+cppcoro::generator<std::string>
+ExportQueryExecutionTrees::computeResultAsQLeverJSONStream(
```

Comment:
For efficiency reasons
(The result has to be packed in reasonably large batches)
the inner generators (like for example this one here)
should not be `generator<string> ` but `stream_generator` (see the other functions).
So the outermost generator of each type
(in particular this one here) should be a `stream_generator`, 
And then best you do your switch for the QLeverJSON also in the single
`computeResultAsStream` function, which then transforms the `stream_generator`
back into a `generator<string> ` which then contains large chunks.
If you want to know more details, this is best explained in a meeting.